tests/fixtures/inbox_data.py

```diff
@@ -167,6 +169,10 @@ def filtered_inbox_data(mlwhdb_test_sessionfactory, qcdb_test_sessionfactory):
         mlwh_db_session.add(well)
     mlwh_db_session.commit()
 
+    # Don't forget to close the sessions
+    qc_db_session.close()
```

Comment:
I guess this is why they recommend `with` blocks, because the session seems to auto-close

Reply:
It looks like the `Session` provider created by `sessionmaker` doesn't close the `Session` if it's used in a with statement. So this doesn't close the session (at least trying it got the other test to hang again)
```python
with mlwhdb_test_sessionfactory() as mlwhdb_session:
    ...  # Do stuff
```

One of [the SQLAlchemy docs](https://docs.sqlalchemy.org/en/13/orm/session_basics.html#when-do-i-construct-a-session-when-do-i-commit-it-and-when-do-i-close-it)'s recommendations is to make a `contextmanager` around the sessionmaker. I could give that a go, but I would need to figure out how that works when it's also a pytest fixture. Maybe having a pytest fixture `yield`ing a Session is the way to go instead...

Reply:
This bit begs to differ: https://docs.sqlalchemy.org/en/14/orm/session_basics.html#using-a-sessionmaker

Reply:
Here's how I was doing it. Can't tell for sure that I don't have the session issue underneath, but could be postgres driver differences again: https://github.com/wtsi-npg/npg_porch/blob/devel/server/npg/porchdb/connection.py#L64

Reply:
It turns out it works as expected 😅. I'm not exactly sure what cause the test to hang again, but it was probably a stray session use not inside a `with` block.
Big `with` blocks wrapping the concerned function bodies work

Reply:
I went with new fixtures with something close to what you did, it requires less indentation in the tests